### PR TITLE
fix: clear obsolete node cache data

### PR DESCRIPTION
**Pull request by giovanecosta**
# TODO
- [ ] Clear current Redis data

### CODE REVIEW

**Review by pedroferreira1 — APPROVED** [review]
The PR for me is good, I was just curious about the ttl as 5 seconds. I feel is too fast to delete some data when the node might just lose some connection for a few seconds. Maybe increase to 30 or 60 seconds. What do you think? @giovanecosta @msbrogli 

Besides that I was thinking about the explorer itself. We keep polling this data when we are seeing a single peer, right? What happens if the peer data is deleted? The page will just become empty?

**Comment by giovanecosta** [comment]
@pedroferreira1 30 seconds sounds fair enough for me.

Very good point on explorer. I've created an issue for that here: https://github.com/HathorNetwork/hathor-explorer/issues/93

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -82,6 +82,7 @@ jobs:
           REDIS_DB: 0
           TOKEN_METADATA_BUCKET: token-metadata-testnet
           CORS_ALLOWED_ORIGIN: https://explorer.testnet.hathor.network
+          NODE_CACHE_TTL: 5
```
Comment (msbrogli) [inline .github/workflows/deploy.yml:85]: 5 seconds? I feel we should wait at least 30 seconds before removing the data.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -96,6 +97,7 @@ jobs:
           REDIS_DB: 0
           TOKEN_METADATA_BUCKET: token-metadata-mainnet
           CORS_ALLOWED_ORIGIN: https://explorer.hathor.network
+          NODE_CACHE_TTL: 5
```
Comment (msbrogli) [inline .github/workflows/deploy.yml:100]: Same thing here. 30 seconds maybe?

File: common/configuration.py [hunk 3]
```diff
@@ -37,6 +37,7 @@ def default(cls) -> 'Environment':
 HATHOR_CORE_DOMAIN = config('HATHOR_CORE_DOMAIN', default=None)
 
 HATHOR_NODES = config('HATHOR_NODES', default='', cast=Csv())
+NODE_CACHE_TTL = config('NODE_CACHE_TTL', default=5)
```
Comment (msbrogli) [inline common/configuration.py:40]: Same here.

File: gateways/node_gateway.py [hunk 4]
```diff
@@ -60,7 +60,7 @@ def save_node(self, id: str, node: Node) -> Union[bool, None]:
         :return: If saved successfuly or not
         :rtype: bool
         """
-        return self.cache_client.set(NODE_COLLECTION_NAME, id, node.to_dict())
+        return self.cache_client.set(NODE_COLLECTION_NAME, id, node.to_dict(), NODE_CACHE_TTL)
```
Comment (msbrogli) [inline gateways/node_gateway.py:63]: This won't fix the entries that are already there, right? Will we do a manual clean up?

**Review by msbrogli — CHANGES_REQUESTED** [review]

File: gateways/node_gateway.py (hunk 4 above)
Comment (giovanecosta) [reply gateways/node_gateway.py:63]: Yes, is in PR's TODO list

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (giovanecosta) [reply .github/workflows/deploy.yml]: Ok, we all agree with 30 then. Changed.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (giovanecosta) [reply .github/workflows/deploy.yml]: changed

File: common/configuration.py (hunk 3 above)
Comment (giovanecosta) [reply common/configuration.py]: changed

**Review by msbrogli — APPROVED** [review]

**Pull request merged**